### PR TITLE
Order backend No frontend

### PR DESCRIPTION
**Pull request by MaxSukhovetskyi**

### CODE REVIEW

**Review by viktormovchaniuk — APPROVED** [review]

**Review by RazSKg — APPROVED** [review]

**Review by VitaliiBondar — APPROVED** [review]

**Review by EvanAntoni — APPROVED** [review]

File: easyrest/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """
+
+    def __init__(self, data, msg=""):
+        self.data = data
+        self.msg = msg
+
+    def __str__(self):
+        base = "Validation Error"
+        base += ": %s " % self.msg
+        base += str(self.data)
```
Comment (serge30) [inline easyrest/exceptions.py:17]: `return "Validation Error: %s %s" % (self.msg, self.data)` is a pythonic way to do such things.

File: easyrest/models/menu_item.py [hunk 2]
```diff
@@ -26,7 +27,11 @@ class MenuItem(Base):
     description = Column(Text)
     ingredients = Column(Text)
     img = Column(Text)
+    price = Column(Float)
```
Comment (serge30) [inline easyrest/models/menu_item.py:30]: Please never use `Float` for money, as it causes some precision lose during arithmetic operations, there is special type ([Numeric](https://docs.sqlalchemy.org/en/latest/core/type_basics.html#sqlalchemy.types.Numeric)) with fixed decimal point there. Do not forget to set correct scale.

File: easyrest/models/meta.py [hunk 3]
```diff
@@ -60,6 +60,11 @@ def as_dict(self):
         """
         data = {}
         for c in self.__table__.columns:
+            if c.name in exclude:
+                continue
+            if len(include) != 0:
+                if c.name not in include:
+                    continue
```
Comment (serge30) [inline easyrest/models/meta.py:67]:
```
if include and c.name not in include:
    continue
```

File: easyrest/models/order.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        return item.as_dict(exclude=exclude, include=include)
+
+    def get_items(self, session, exclude=[], include=[]):
+        model = self.items
+        items = []
+        for item in model:
+            food = item.food
+            d = food.as_dict(exclude=exclude, include=include)
+            d.update({"quantity": item.quantity})
+            d.update({"q_id": item.id})
```
Comment (serge30) [inline easyrest/models/order.py:58]: Combine two statements in one:
```
d.update({"quantity": item.quantity, "q_id": item.id})
```

File: easyrest/models/order.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        item.quantity = value
+
+        return item
+
+    def count_total(self):
+        total = 0
+        for item in self.items:
+            q = item.quantity
+            p_per_item = item.food.price if item.food.price is not None else 3.50
```
Comment (serge30) [inline easyrest/models/order.py:100]: What the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) here?

File: easyrest/views/order_controller.py [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+        raise HTTPForbidden("Wrong input data %s" % e)
+
+    order = get_order(request, order_id)
+
+    status, role, action = order.status, request.token.user.role.name, json["action"]
+
+    # State transition graph
+    # (status, role, action)
+    # User->Submit = User press Submit
+    grahp = {
```
Comment (serge30) [inline easyrest/views/order_controller.py:338]: It's a good candidate for constant.

**Pull request merged**